Move all assets to test-support.js instead of vendor.js

simonihmig opened the pull request:
This fixes #397 and also implements the other changes suggested there: All (own and imported) JS is moved from `vendor.js` to `test-support.js`. The repo's code is now in the `addon-test-support` tree, and the trickery in `treeForAddonTestSupport()` (shamelessly stolen from `ember-native-dom-helpers`) makes sure the imports have not changed (i.e. no `test-support/` path added to imports).

I tried this fork with a large app, and was working fine. Didn't have to update anything (except `package.json`), and all still working. I hope I am not missing something though, so please double check if these changes make sense for you! :)

san650 commented:
Thanks @simonihmig for working on this!

As you said we have to test this change to validate there are not regressions.

I'll test it on some small projects I have but it would be great if @magistrula and others that have big test suites can test it as well.

ro0gr commented:
@simonihmig I've just [merged](https://github.com/san650/ember-cli-page-object/pull/396) docs updates which produced conflicts in your PR. Sorry for not merging it earlier. Can you please resolve conflicts?

ro0gr commented:
Hm.. Seems like `addon-test-support/` is not included in the code coverage reports.

simonihmig commented on index.js:
```diff
@@ -21,39 +21,53 @@ module.exports = {
+    });
+  },
+
+  _shouldIncludeFiles() {
```
Yeah, good catch. Makes sense!

simonihmig commented:
Thanks for the quick feedback!

I rebased this, and cleaned this up a bit further.
Removed `_shouldIncludeFiles`, and changed the node-assets config a bit, as it does not need to be dynamic anymore. This revealed that `ceibo` was still imported to `vendor.js`, which should be fixed now as well.

> I believe we still have to change API paths for the docs generator as well

I updated this line, but as I am not aware how that works, please double check...

> I'll test it on some small projects I have but it would be great if @magistrula and others that have big test suites can test it as well.

Absolutely! I reran the latest changes on one of our apps, and its 1700+ tests (all included) are still all passing.

> Hm.. Seems like addon-test-support/ is not included in the code coverage reports.

Sorry, I have no idea how that works! 😝 Also wondered how coverage could have changed significantly. Anything I can do on my side to fix this?

ro0gr commented:
> Also wondered how coverage could have changed significantly. Anything I can do on my side to fix this?

Looks like it should be [fixed](https://github.com/kategengler/ember-cli-code-coverage/pull/160) in the latest of ember-cli-code-coverage. I think we just have to upgrade it.

ro0gr commented:
> I think we just have to upgrade it.

I've just tried `ember-cli-code-coverage@1.0.0-beta.3` locally. Unfortunatelly it reports zero coverage for each of file in the `addon-test-support/` folder. Probably it's related to the import path tricks borrowed from `ember-native-dom-helpers`.

ro0gr commented:
@simonihmig FYI I'm currently working on the code coverage issue on top of your branch. The issue is caused by the fact that `ember-cli-code-coverage` expects generated code to have a fixed module path ([`addon-name/test-support`](https://github.com/kategengler/ember-cli-code-coverage/blob/77b5b9201060d634d281f64a5bb7b412773a8b89/index.js#L178)). So the possible solution is to change only entries module names(`index`, `extend`, `macros`) and leave the rest of the files under `ember-cli-page-object/test-support` namespaced.
 
wip branch: https://github.com/ro0gr/ember-cli-page-object/commits/addon-test-support-coverage

simonihmig commented:
@ro0gr Hm, I see. Thanks for working on this, and for giving feedback! Hope it works out somehow!

ro0gr commented:
@simonihmig just merged #400. Thanks for the great idea!
